corehq/apps/hqwebapp/templates/hqwebapp/base.html

```diff
@@ -407,19 +408,41 @@
             {# do not override this block, use registerurl template tag to populate #}
         {% endblock %}
         </div>
+        {% if request|toggle_enabled:"ANALYTICS_NEW" or USE_NEW_ANALYTICS %}
+        {% include 'analytics/initial/all.html' %}
+        <div class="initial-analytics-data hide">
+        {% block initial_analytics_data %}
+            {# do not override this block, use initial_analytics_data template tag to populate #}
+        {% endblock %}
+        </div>
+        <div class="analytics-ab-tests hide">
+        {% block analytics_ab_test %}
+            {# do not override this block, use analytics_ab_test template tag to populate #}
+        {% endblock %}
+        </div>
+        {% endif %}
 
         {# javascript below this line #}
 
         {# HQ Specific Libraries #}
         <script src="{% statici18n LANGUAGE_CODE %}"></script> {# DO NOT COMPRESS #}
+
+        {# Core dependencies for analytics & HQ's libraries #}
+        {% compress js %}
+        <script src="{% static 'hqwebapp/js/hqModules.js' %}"></script>
+        <script src="{% static 'hqwebapp/js/initial_page_data.js' %}"></script>
+        {% endcompress %}
+
+        {% if request|toggle_enabled:"ANALYTICS_NEW" or USE_NEW_ANALYTICS %}
+        {% include 'analytics/analytics_js.html' %}
```

Comment:
Can this just be included after the compress block below, around line 424? Or are there scripts in that block that depend on analytics?

I'm a little worried about what we define as "core" HQ libraries and about the dependencies among them being fragile (this has come up in the RequireJS work, which also makes some changes to this set of script tags).

Reply:
there are indeed scripts in that block that depend on analytics.
`hq.helpers`
`main.js`

Reply:
Can you comment those dependencies in this file?